overlord/state/state.go

```diff
@@ -92,12 +92,13 @@ type State struct {
 	// for registering runtime callbacks
 	lastHandlerId int
 
-	backend  Backend
-	data     customData
-	changes  map[string]*Change
-	tasks    map[string]*Task
-	warnings map[string]*Warning
-	notices  map[noticeKey]*Notice
+	backend        Backend
+	data           customData
+	changes        map[string]*Change
+	tasks          map[string]*Task
+	warnings       map[string]*Warning
+	notices        map[noticeKey]*Notice
+	noticeLastDate time.Time
```

Comment:
I believe this needs to persisted, please check `State.MarshalJSON()` and `State.UnmarshalJSON()`.

Reply:
Good point, maybe we don't need to persist it.

CC: @pedronis

Reply:
My view is its dangerous to rely on time elapsing/delays to guarantee correct behaviour. I would play it safe unless there is massive downside to consider.

Reply:
I agree. So, since persistence is already implemented, I'll close this.